src/components/EditActivityPage/EditActivityPage.js

```diff
@@ -35,28 +55,44 @@ const EditActivityPage = ({ activity, history, onStartEditActivity, onStartRemov
     }
   }, [activityId]);
 
+  useEffect(() => {
+    (async () => {
```

Comment:
it will be better if you will assign function to variable and invoke this function. More readable, but still nice job ;)